### PR TITLE
Always set FieldWithLabel id

### PR DESCRIPTION
**Pull request by jeanniefu**
### Description
Always set FieldWithLabel id. If missing, generates a random one.

We used to always generate a random ID if one was not set, but this was accidentally removed due to some refactoring in #2281.

One known issue this causes is that labels are no longer linked to inputs, which is a problem for screen readers.

## Release notes:

Fix screenreader bug where labels and inputs are not linked with ID.

### Checklist

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

#2388

### CODE REVIEW

File: server/app/views/components/FieldWithLabel.java [hunk 1]
```diff
@@ -317,7 +317,6 @@ private DivTag getNonNumberInputTag() throws RuntimeException {
 
   /** Public final tag getters * */
   public DivTag getTextareaTag() throws RuntimeException {
-    genRandIdIfEmpty();
```
Comment (clouser2) [inline server/app/views/components/FieldWithLabel.java:320]: Why remove this call?

**Review by clouser2 — APPROVED** [review]

**Comment by clouser2** [comment]
Thanks for fixing!

File: server/app/views/components/FieldWithLabel.java (hunk 1 above)
Comment (jeanniefu) [reply server/app/views/components/FieldWithLabel.java:320]: It'll be called instead in the shared applyAttrsAndGenLabel(...) function below. That shared function (or checkboxApplyAttrsAndGenLabel) will be called by all the public getXTag specific functions.

Comment (clouser2) [reply server/app/views/components/FieldWithLabel.java:320]: Aha, I'd totally missed that one. Also, the addition of so many tests gave me a lot of confidence that it worked, so thanks for that as well :).

**Pull request merged**